rockcraft/pebble.py

```diff
@@ -0,0 +1,84 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Pebble metadata and configuration helpers."""
+
+import glob
+from pathlib import Path
+
+import yaml
+from craft_cli import emit
+
+
+class Pebble:
+    """Handles the Pebble metadata information and setup operations."""
+
+    PEBBLE_PATH = "var/lib/pebble/default"
+    PEBBLE_LAYERS_PATH = f"{PEBBLE_PATH}/layers"
+    PEBBLE_BINARY_PATH = "bin/pebble"
+    PEBBLE_PART_SPEC = {
+        "plugin": "nil",
+        "stage-snaps": ["pebble/latest/edge"],
+        "stage": [PEBBLE_BINARY_PATH],
+        "override-prime": f"craftctl default\nmkdir -p {PEBBLE_LAYERS_PATH}",
```

Comment:
So if the actual pebble layer file is written outside of the parts lifecycle, in `define_pebble_layer()`, is there a need for this part to create this path? Doesn't `define_pebble_layer()` do it too, on line 73?

Reply:
that sounds accurate since line 74 is using `parents=True`. Changing it

Reply:
so how come this stayed? Was there a problem?

Reply:
Yes, I forgot (again lol) that `services` is optional. So if there are no `services`, the `set_pebble_services` will not run, and this path won't exist, which will make `pebble enter` complain about it. we could set `--create-dirs` within the OCI entrypoint, but this is cleaner.